### PR TITLE
Finish skeleton code for application

### PR DESCRIPTION
**Pull request by timkral**
This is the last of the skeleton code for Scala. There are three new modules:
1. Store - Similar to our idea before. Contains an interface called `Store` to do basic CRUD. And include a skeleton implementation for Hyperdex.
2. Model - A place to keep stuff about our models. Right now there's only a single class called `Entity` in there.
3. API - A place for our RESTful resources. I added one for `/about` and `/teams`

### CODE REVIEW

File: api/src/main/scala/net/spals/leagur/api/AboutResource.scala [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+/**
+ * RESTful API resource for showing server information
+ *
+ * @author tkral
+ */
+@Path("leagur/about")
+@Produces(Array(MediaType.APPLICATION_JSON))
+@Consumes(Array(MediaType.APPLICATION_JSON))
+@AutoBindSingleton
```
Comment (timkral) [inline api/src/main/scala/net/spals/leagur/api/AboutResource.scala:17]: API resources get auto bound via governator

File: api/src/main/scala/net/spals/leagur/api/TeamsResource.scala [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+/**
+ * RESTful API resource for teams
+ *
+ * @author tkral
+ */
+@Path("leagur/teams")
+@Produces(Array(MediaType.APPLICATION_JSON))
+@Consumes(Array(MediaType.APPLICATION_JSON))
+@AutoBindSingleton
+case class TeamsResource @Inject() (store: Store) {
```
Comment (timkral) [inline api/src/main/scala/net/spals/leagur/api/TeamsResource.scala:20]: We use guice to inject dependencies

File: app/src/main/scala/net/spals/leagur/app/LeagurApp.scala [hunk 3]
```diff
@@ -77,6 +78,10 @@ class LeagurApp(serviceFileConfigName: String) extends Application[Configuration
         LeagurApp.LOGGER.info("Shutting down Leagur application...")
         manager.close
       }
+
+      val injector = lifecycleInjector.createInjector
+      env.jersey().register(injector.getInstance(classOf[AboutResource]))
+      env.jersey().register(injector.getInstance(classOf[TeamsResource]))
```
Comment (timkral) [inline app/src/main/scala/net/spals/leagur/app/LeagurApp.scala:84]: Register API resources with DropWizard

File: store/src/main/scala/net/spals/leagur/store/hyperdex/impl/HyperdexStore.scala [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+import scala.collection.JavaConverters._
+
+/**
+ * Hyperdex implementation of {@link Store}
+ *
+ * @author tkral
+ */
+@AutoBindSingleton(classOf[Store])
+class HyperdexStore extends Store {
```
Comment (timkral) [inline store/src/main/scala/net/spals/leagur/store/hyperdex/impl/HyperdexStore.scala:17]: A skeleton for the HyperDex implementation of `Store`. We'll need to fill this out, which I'd imagine isn't that hard. Also need to work on some configuration injection so we know how to contact the HyperDex coordinator.

Comment (thespags) [inline store/src/main/scala/net/spals/leagur/store/hyperdex/impl/HyperdexStore.scala:17]: Perhaps we will add an InMemoryStore for testing. I've found that useful when writing some db layered things.

**Pull request merged**